src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -933,7 +933,45 @@ public static bool IsPointerType(this TypeSymbol type)
 
         internal static int FixedBufferElementSizeInBytes(this TypeSymbol type)
         {
-            return type.SpecialType.FixedBufferElementSizeInBytes();
+            var specialType = type.SpecialType;
+
+            if (specialType != SpecialType.None)
+            {
+                return specialType.FixedBufferElementSizeInBytes();
+            }
+
+            if (type.TypeKind != TypeKind.Struct)
+            {
+                // TODO: Add Diagnostic
+                return 0;
+            }
+
+            var symbols = type.GetMembers().WhereAsArray((symbol) => symbol is SourceMemberFieldSymbol);
```

Comment:
And does this skip backing fields? Will this method return the correct result for:
```
struct S
{
    public int F;
    public int P { get; set; }
}
```